unity/Assets/UnitTests/TestThirdPartyCamera.cs

```diff
@@ -0,0 +1,106 @@
+
+namespace Tests {
+    public class TestThirdPartyCamera : TestBase {
+        [UnityTest]
+        public IEnumerator TestAddThirdPartyCamera() {
+            Dictionary<string, object> action = new Dictionary<string, object>();
+
+            action["action"] = "Initialize";
+            action["fieldOfView"] = 90f;
+            action["snapToGrid"] = true;
+            yield return step(action);
+
+            action.Clear();
+
+            action["action"] = "AddThirdPartyCamera";
+            action["position"] = Vector3.zero;
+            action["rotation"] = Vector3.zero;
+            action["orthographic"] = true;
+            action["orthographicSize"] = 5;
+            yield return step(action);
+
+            Assert.NotNull(GameObject.Find("ThirdPartyCamera0"));
+        }
+
+        [UnityTest]
+        public IEnumerator TestAddMultipleThirdPartyCamera() {
+            Dictionary<string, object> action = new Dictionary<string, object>();
+
+            action["action"] = "Initialize";
+            action["fieldOfView"] = 90f;
+            action["snapToGrid"] = true;
+            yield return step(action);
+
+            action.Clear();
+
+            action["action"] = "AddThirdPartyCamera";
+            action["position"] = Vector3.zero;
+            action["rotation"] = Vector3.zero;
+            action["orthographic"] = true;
+            action["orthographicSize"] = 5;
+            yield return step(action);
+            yield return step(action);
+
+            Assert.NotNull(GameObject.Find("ThirdPartyCamera0"));
+            Assert.NotNull(GameObject.Find("ThirdPartyCamera1"));
+
+        }
+
+        [UnityTest]
+        public IEnumerator TestCoordinateFromRaycastThirdPartyCamera() {
+            Dictionary<string, object> action = new Dictionary<string, object>();
+
+            action["action"] = "Initialize";
+            action["fieldOfView"] = 90f;
+            action["snapToGrid"] = true;
+            yield return step(action);
+
+            action.Clear();
+
+            action["action"] = "AddThirdPartyCamera";
+            action["position"] = Vector3.zero;
+            action["rotation"] = Vector3.zero;
+            action["orthographic"] = true;
+            action["orthographicSize"] = 5;
+            yield return step(action);
+
+            action.Clear();
+
+            //test raycast when hitting something in scene
+            action["action"] = "CoordinateFromRaycastThirdPartyCamera";
+            action["x"] = 0.5f;
+            action["y"] = 0.5f;
+            action["thirdPartyCameraId"] = 0;
+            yield return step(action);
+
+            bool result = false;
+            Vector3 coord = (Vector3)actionReturn;
+
+            result = Mathf.Approximately(coord.x, 0.0f);
+            Assert.AreEqual(result, true);
+
+            result = Mathf.Approximately(coord.y, 0.0f);
+            Assert.AreEqual(result, true);
+
+            result = Mathf.Approximately(coord.z, 2.472f);
```

Comment:
What is it hitting here? how did you get then number?
This is why I think in the future we should also have tests use simple procedural scenes for tests since it's not obvious this should be the behavior from reading the test.

Reply:
yeah this value is based on spawning a camera into FloorPlan_1, so if FloorPlan_1 changes it could invalidate this test. Some sort of procedural test scene should be the way to go in the future, but currently the tests default initializing and loading up FloorPlan_1. I got the value from a direct debug log of the return of the action. We currently use similar values obtained directly from the console in things like the `PutObject` test.